In [4]:
# Import Pandas for data analysis
import pandas as pd

# Check the installed Pandas version
print(pd.__version__)

3.0.5


### Pandas version 3.0.5

I imported Pandas because it will be used to load, clean and analyse
the healthcare insurance dataset.

I also checked the installed Pandas version to record the software
environment used for this project.

### Loading the dataset

I loaded the healthcare insurance CSV file into a Pandas DataFrame
called `df`. A DataFrame organises data into rows and columns,
making it suitable for data exploration and analysis.

In [5]:
# Load the healthcare insurance dataset
df = pd.read_csv("insurance.csv")

### Initial preview of the dataset

The first five rows were displayed using `df.head()` to confirm that
the dataset was loaded successfully. The dataset contains information
about age, sex, BMI, number of children, smoking status, geographic
region and medical insurance charges.

The preview provides an initial understanding of the variables that
will be investigated during the analysis.

In [ ]:
# Display the first five rows of the dataset

df.head()

,age,sex,bmi,children,smoker,region,charges
0,19,female,27.900,0,yes,southwest,16884.92400
1,18,male,33.770,1,no,southeast,1725.55230
2,28,male,33.000,3,no,southeast,4449.46200
3,33,male,22.705,0,no,northwest,21984.47061
4,32,male,28.880,0,no,northwest,3866.85520


### Understanding the DataFrame

The `df.info()` method is used to get a summary of the DataFrame. It shows the number of rows and columns, the column names, the number of non-null values, and the data type of each column.

The dataset contains 1,338 rows and 7 columns. All columns have 1,338 non-null values, which means there are no missing values in the dataset.

The data types show that `age` and `children` are integers, `bmi` and `charges` are floating-point numbers, and `sex`, `smoker`, and `region` contain text data.

This initial check helps identify the structure of the dataset and whether there are any missing values or data types that may need to be addressed during data cleaning.


In [6]:
# Display information about the dataset structure and data types

df.info()

<class 'pandas.DataFrame'>
RangeIndex: 1338 entries, 0 to 1337
Data columns (total 7 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   age       1338 non-null   int64  
 1   sex       1338 non-null   str    
 2   bmi       1338 non-null   float64
 3   children  1338 non-null   int64  
 4   smoker    1338 non-null   str    
 5   region    1338 non-null   str    
 6   charges   1338 non-null   float64
dtypes: float64(2), int64(2), str(3)
memory usage: 73.3 KB


### Checking for Missing Values

Missing values can affect the accuracy of data analysis and may need to be handled before analysing the dataset.

I will check each column for missing values to determine whether any records require further cleaning.

Though df.info() checks and tells about missing data, i am still running the missing value code as the data cleaning process.

The `isnull().sum()` method counts the number of missing values in each column.


In [8]:
df.isnull().sum()

age         0
sex         0
bmi         0
children    0
smoker      0
region      0
charges     0
dtype: int64

**The result shows that there are no missing values in the dataset, so no missing data needs to be filled or removed.**

### Checking for Duplicate Rows

The `df.duplicated().sum()` method is used to check whether the DataFrame contains any completely duplicated rows.

The result shows that there is 1 duplicate row in the dataset. Identifying duplicate records is an important part of data cleaning because duplicate rows can affect the accuracy of the analysis.


In [7]:
df.duplicated().sum()

np.int64(1)

### Inspecting the Duplicate Row

The `df[df.duplicated()]` method is used to display the rows that have been identified as duplicates.

This allows me to inspect the duplicated record before deciding whether it should be removed from the dataset.


In [11]:
df[df.duplicated()]

,age,sex,bmi,children,smoker,region,charges
581,19,male,30.59,0,no,northwest,1639.5631



### Confirming Duplicate Records

The `df[df.duplicated(keep=False)]` method is used to display all rows that are part of a duplicate, including the original and the duplicated record.

Using `keep=False` allows both identical rows to be displayed so that I can confirm that the records are complete duplicates before deciding whether to remove them.

This provides an additional check before making changes to the dataset.


In [14]:
df[df.duplicated(keep=False)]

,age,sex,bmi,children,smoker,region,charges
195,19,male,30.59,0,no,northwest,1639.5631
581,19,male,30.59,0,no,northwest,1639.5631


### Removing the Duplicate Row

The inspection confirms that the records at index 195 and index 581 contain exactly the same values. Therefore, one of the duplicate records can be removed to prevent the duplicate data from affecting the analysis.

The `drop_duplicates()` method is used to remove duplicate rows from the DataFrame.


In [15]:
df = df.drop_duplicates()

### Verifying Duplicate Removal

After removing the duplicate row, I will check the DataFrame again to confirm that no duplicate rows remain.

The `df.duplicated().sum()` method is used to count the remaining duplicate rows. A result of 0 will confirm that the duplicate has been successfully removed.


In [16]:
df.duplicated().sum()

np.int64(0)

### Checking Categorical Values

The categorical columns contain text values rather than numerical measurements. I will check the unique values in these columns to identify any unexpected or inconsistent entries that may need cleaning.

This check helps confirm that the values are consistent and suitable for analysis.


In [17]:
df['sex'].unique()

<StringArray>
['female', 'male']
Length: 2, dtype: str

 **The sex column contains two consistent values, female and male, so no cleaning is required for this column.**

 **Next I will check the unique values in the smoker column to identify any unexpected or inconsistent entries.**

In [18]:
df['smoker'].unique()

<StringArray>
['yes', 'no']
Length: 2, dtype: str

 **The smoker column contains only yes and no, so no cleaning is required.**

In [19]:
df['region'].unique()

<StringArray>
['southwest', 'southeast', 'northwest', 'northeast']
Length: 4, dtype: str

 **The region column contains four consistent values, so no cleaning is required.**

### Checking Age Values

**The `age` column contains numerical values. I will check the minimum and maximum age to identify any values that may be outside a sensible range or require further investigation.**


In [20]:
df['age'].min(), df['age'].max()

(np.int64(18), np.int64(64))

**The age values range from 18 to 64, which is a sensible range for the adult insurance records in this dataset, so no cleaning is required and no obviously invalid age values were identified. Therefore, no cleaning is required for the age column.**


### Checking BMI Values

**The `bmi` column contains numerical values. I will check the minimum and maximum BMI values to identify any values that may be outside a plausible range or require further investigation.**


In [21]:
df['bmi'].min(), df['bmi'].max()

(np.float64(15.96), np.float64(53.13))

 **The BMI values range from 15.96 to 53.13. These are plausible recorded BMI values, so no obviously invalid values were identified and no cleaning is required for the bmi column.**

### Checking Children Values

**The children column contains numerical values. I will check the minimum and maximum number of children to identify any values that may be invalid or require further investigation.**


In [22]:
df['children'].min(), df['children'].max()

(np.int64(0), np.int64(5))


**The number of children ranges from 0 to 5. These are plausible values, so no obviously invalid values were identified and no cleaning is required for the children column.**




### Checking Charges Values

**The `charges` column contains the insurance charges as numerical values. I will check the minimum and maximum charges to identify any values that may be invalid or require further investigation.**


In [23]:
df['charges'].min(), df['charges'].max()

(np.float64(1121.8739), np.float64(63770.42801))

**The insurance charges range from 1,121.8739 to 63,770.42801. All values are positive, so no obviously invalid values were identified and no cleaning is required for the `charges` column.**



### Descriptive Statistics

df.describe() is a Pandas method for descriptive statistics, it gives a statistical summary of my numerical columns.
Descriptive statistics provide a summary of the numerical columns in the dataset. They help us understand the distribution and general characteristics of variables such as age, BMI, number of children, and insurance charges.


In [6]:
df.describe()

,age,bmi,children,charges
count,1338.000000,1338.000000,1338.000000,1338.000000
mean,39.207025,30.663397,1.094918,13270.422265
std,14.049960,6.098187,1.205493,12110.011237
min,18.000000,15.960000,0.000000,1121.873900
25%,27.000000,26.296250,0.000000,4740.287150
50%,39.000000,30.400000,1.000000,9382.033000
75%,51.000000,34.693750,2.000000,16639.912515
max,64.000000,53.130000,5.000000,63770.428010


**The descriptive statistics provide an overview of the numerical variables. After removing the duplicate record, the dataset contains 1,337 records. The summary shows the central values and ranges of age, BMI, children, and insurance charges, which will help guide the next stage of the analysis.**
